plugin/src/main/java/uk/co/deliverymind/wiremock/maven/plugin/WireMockMojo.java

```diff
@@ -2,17 +2,47 @@
 
 import com.github.tomakehurst.wiremock.standalone.WireMockServerRunner;
 import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.codehaus.plexus.classworlds.realm.ClassRealm;
 
+import java.io.File;
+import java.net.MalformedURLException;
 import java.util.Arrays;
+import java.util.List;
 
-@Mojo(name = "run", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST)
+@Mojo(name = "run", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST, requiresDependencyResolution = ResolutionScope.TEST)
 public class WireMockMojo extends ConfigurationMojo {
 
+    @Parameter(defaultValue = "${plugin}", required = true, readonly = true)
+    private PluginDescriptor descriptor;
+
+    @Parameter(property = "project.testClasspathElements", required = true, readonly = true)
+    private List<String> classpathElements;
+
     public void execute() throws MojoExecutionException {
+        loadTestClasspath();
```

Comment:
I have some doubts if enabling extensions by default is the best thing to do in this plugin. I can think og an extra param in pom to manage this. However, let's keep it as it is for now. If it starts causing troubles for other users we can change it.